esmvalcore/preprocessor/_regrid.py

```diff
@@ -252,15 +252,64 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
                 [coord] = coords
                 cube.remove_coord(coord)
 
-    # Perform the horizontal regridding.
-    if _attempt_irregular_regridding(cube, scheme):
-        cube = esmpy_regrid(cube, target_grid, scheme)
-    else:
-        cube = cube.regrid(target_grid, HORIZONTAL_SCHEMES[scheme])
+    # Return unaltered cube if hotizontal grid is the same.
```

Comment:
Note that the cube is not necessarily unaltered at this point, because the code above removes certain coordinates. Maybe this check should go before that?

Reply:
maybe call it "unregridded". It is not going back to before what was done in this method, you are right.